simperby/src/lib.rs

```diff
@@ -86,9 +92,81 @@ pub struct ConsensusVoteItem {
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub struct GenesisInfo {
     pub header: BlockHeader,
+    pub genesis_signature: FinalizationProof,
     pub chain_name: String,
 }
 
+impl GenesisInfo {
+    fn create_genesis_block(&self) -> Block {
+        Block {
+            header: self.header.clone(),
+            transactions: vec![],
+        }
+    }
+}
+
+/// A set of operations that may update the Simperby node state.
+///
+/// `ProposeBlock` and `SubmitConsensusVote` are explicitly performed through the `SimperbyApi` trait, and the others
+/// are done implicitly by the background task which is triggered by incoming p2p network messages.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub enum SimperbyOperation {
+    ProposeBlock {
+        block: BlockHeader,
+        signature: TypedSignature<BlockHeader>,
+    },
+    SubmitConsensusVote {
+        hash: Hash256,
+        signature: TypedSignature<BlockHeader>,
```

Comment:
What block this `BlockHeader` indicates?

Reply:
Interface updated. Please take a look.